spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFacforyTests.java

```diff
@@ -89,4 +95,62 @@ public void shouldReturnNetFlavor() throws Exception {
 		assertThat(client instanceof TcpNetClientConnectionFactory).isTrue();
 	}
 
+	@Test
+	void netCustomServer() {
+		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
+		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
+		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
+		TcpNetServerConnectionFactory server = Tcp.netServer(0)
+				.tcpSocketSupport(sockSupp)
+				.tcpNetConnectionSupport(conSupp)
+				.tcpSocketFactorySupport(factSupp)
+				.get();
+		assertThat(TestUtils.getPropertyValue(server, "tcpSocketSupport")).isSameAs(sockSupp);
+		assertThat(TestUtils.getPropertyValue(server, "tcpNetConnectionSupport")).isSameAs(conSupp);
+		assertThat(TestUtils.getPropertyValue(server, "tcpSocketFactorySupport")).isSameAs(factSupp);
+	}
+
+	@Test
+	void nioCustomServer() {
+		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
+		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
+		TcpNioServerConnectionFactory server = Tcp.nioServer(0)
+				.tcpSocketSupport(sockSupp)
+				.usingDirectBuffers(true)
+				.tcpNioConnectionSupport(conSupp)
+				.get();
+		assertThat(TestUtils.getPropertyValue(server, "tcpSocketSupport")).isSameAs(sockSupp);
+		assertThat(TestUtils.getPropertyValue(server, "usingDirectBuffers", Boolean.class)).isTrue();
+		assertThat(TestUtils.getPropertyValue(server, "tcpNioConnectionSupport")).isSameAs(conSupp);
+	}
+
+	@Test
+	void netCustomClient() {
+		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
+		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
+		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
+		TcpNetClientConnectionFactory client = Tcp.netClient("localhost", 0)
+				.tcpSocketSupport(sockSupp)
+				.tcpNetConnectionSupport(conSupp)
+				.tcpSocketFactorySupport(factSupp)
+				.get();
+		assertThat(TestUtils.getPropertyValue(client, "tcpSocketSupport")).isSameAs(sockSupp);
+		assertThat(TestUtils.getPropertyValue(client, "tcpNetConnectionSupport")).isSameAs(conSupp);
+		assertThat(TestUtils.getPropertyValue(client, "tcpSocketFactorySupport")).isSameAs(factSupp);
+	}
+
+	@Test
+	void nioCustomClient() {
+		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
+		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
+		TcpNioClientConnectionFactory client = Tcp.nioClient("localhost", 0)
+				.tcpSocketSupport(sockSupp)
+				.usingDirectBuffers(true)
+				.tcpNioConnectionSupport(conSupp)
```

Comment:
Cannot we call methods as simple as `socketSupport`, `connectionSupport` and `socketFactorySupport`?
We are really in the context of ` Tcp.netClient` builder or `Tcp.nioClient`, so it sounds redundant for me to mention `tcp` or `nio` in every method of the builder again.